### ISSUE TITLE
Unable to access single player mode on Gears of War

### ISSUE THREAD

**Issue by CorpseSlayer** [xenia-canary/xenia-canary#255]
### Validation

- [X] I've read the [FAQ](https://github.com/xenia-project/xenia/wiki/FAQ).
- [X] The Xenia build used is from the master branch (not MLBS/AlexVS/Canary/pull requests, etc.)
- [X] This issue isn't for tech support (help with Xenia).
- [X] If this issue occurs in a specific game, I've done analysis to locate the faulty subsystem of the emulator and a potential reason in it.
- [X] I've checked if this issue hasn't already been reported.
- [X] My device meets the minimum requirements: https://github.com/xenia-project/xenia/wiki/Quickstart#system-requirements
- [ ] (If building) I have read the building doc: https://github.com/xenia-project/xenia/blob/master/docs/building.md

### Describe what's going wrong

I been trying to play Gears of War but it keeps saying "to many profiles connect " this is annoying since there no where that says how to disable the second profile...

### Describe what should happen

I should be able to access single player with 1 profile but when i press Y it logs 2 profiles instead of 1
<img width="961" alt="Screenshot 2023-12-20 194304" src="https://github.com/xenia-canary/xenia-canary/assets/126618918/75e8fb47-45f7-4a8b-a31d-47e9ad5147af">

### If applicable, provide a callstack here, especially for crashes

N/A

### If applicable, upload a logfile and link it here

N/A

**Comment by antoniodesousa** [comment]
I can't reproduce your issue. It's working fine for me.

**Comment by CorpseSlayer** [comment]
> I can't reproduce your issue. It's working fine for me.

i have no idea on to why you can't reproduce this, is very simple Xenia canary will load Gears of War but you wont be able to save the game if you are not sign in, if you press Y to sign in 2 profiles will sign in making the game  impossible to start in 1 player mode since Xenia canary loads 2 profiles instead of 1 profile!
![Screenshot 2024-01-24 000219](https://github.com/xenia-canary/xenia-canary/assets/126618918/d7fc3006-e6f0-432b-8818-f520c1d19907)
![Screenshot 2024-01-23 235629](https://github.com/xenia-canary/xenia-canary/assets/126618918/049a0544-6934-4042-a364-e333d22a43e7)

**Comment by Gliniak** [comment]
Strange that it doesn't load default profile. However fact that pressing Y causes second profile to pop means you have 2 or more controllers connected. If you use any virtual controller then it might cause this issue.

**Comment by CorpseSlayer** [comment]
> Strange that it doesn't load default profile. However fact that pressing Y causes second profile to pop means you have 2 or more controllers connected. If you use any virtual controller then it might cause this issue.

That's the weird issue i only have 1 controller connected to my PC, i can tell you what i am using
i have a Series S controller with Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X (https://www.mayflash.com/product/magic_x.html) i have nothing else connected to my PC..

**Comment by antoniodesousa** [comment]
> Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X

Most likely that is the issue. Somehow is making Xenia thing that there're 2 controllers plugin.
Can you try with a different adapter or PC?

**Comment by CorpseSlayer** [comment]
> > Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X
> 
> Most likely that is the issue. Somehow is making Xenia thing that there're 2 controllers plugin. Can you try with a different adapter or PC?

MAYFLASH Magic NS Wireless Bluetooth Controller adapter this one is a different version 
of the one i am using right now is for other controller to connect to PC (does not includes Series S controller)
i do not use wired controllers since i drop every console i have had by pulling the cable of the controller....lol

Test with MAYFLASH Magic NS Wireless Bluetooth Controller adapter the result are good
![Screenshot 2024-01-24 142833](https://github.com/xenia-canary/xenia-canary/assets/126618918/7122e871-a11d-46da-90ea-b61a5ab01595)

For some weird reason Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X is not working with Xenia Canary 
is there anyway you can make this adapter work some how with Xenia Canary?

**Comment by Gliniak** [comment]
Set log_level to 3 and send log from it. I suspect changing HID in config to Xinput or SDL might help

**Comment by CorpseSlayer** [comment]
> Set log_level to 3 and send log from it. I suspect changing HID in config to Xinput or SDL might help

i will try this to if this fixes anything...

**Comment by CorpseSlayer** [comment]
> Set log_level to 3 and send log from it. I suspect changing HID in config to Xinput or SDL might help

So i did a  couple of test here are the results:

Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X Alone with  HID change from Any to SDL make USER profile load

MAYFLASH Magic NS Wireless Bluetooth Controller adapter Alone with  HID on Any loads USER profile

Now if any of these 2 get connected to the PC at the same time then Xenia wont recognize the Default USER profile...

I will have to upload the log file some where else is 119mb is mediafire website ok?

**Comment by Gliniak** [comment]
> Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X Alone
> MAYFLASH Magic NS Wireless Bluetooth Controller adapter Alone

You have 2 adapters? That what probably causes issues.

You can zip log and then it will be small enough. Plus I only need log from like first 5 seconds after you load the game

**Comment by CorpseSlayer** [comment]
> > Magic-X Wireless Bluetooth USB Adapter for Xbox Series S/X Alone
> > MAYFLASH Magic NS Wireless Bluetooth Controller adapter Alone
> 
> You have 2 adapters? That what probably causes issues.
> 
> You can zip log and then it will be small enough. Plus I only need log from like first 5 seconds after you load the game

Yes  I have  2 but not connected at the same time i have many controllers most of the time i use the Magic-X, the NS adapter i use if my friends visit since connecting the 2 adapter has never gave me problems when using them in other emulators here:
[xenia.zip](https://github.com/xenia-canary/xenia-canary/files/14042974/xenia.zip)

**Comment by Gliniak** [comment]
There is your answer: 
`i> 00004480 SDL OnControllerDeviceAdded: "Controller (MAGIC-NS)", JoystickType(1), GameControllerType(1), VendorID(0x0079), ProductID(0x18D3)
i> 00004480 SDL OnControllerDeviceAdded: Added at index 0.
i> 00004480 SDL OnControllerDeviceAdded: "Xbox 360 Controller", JoystickType(1), GameControllerType(1), VendorID(0x045E), ProductID(0x028E)
i> 00004480 SDL OnControllerDeviceAdded: Added at index 1.`

Magic-Ns adapter is itself visible as a controller

**Comment by CorpseSlayer** [comment]
> There is your answer: `i> 00004480 SDL OnControllerDeviceAdded: "Controller (MAGIC-NS)", JoystickType(1), GameControllerType(1), VendorID(0x0079), ProductID(0x18D3) i> 00004480 SDL OnControllerDeviceAdded: Added at index 0. i> 00004480 SDL OnControllerDeviceAdded: "Xbox 360 Controller", JoystickType(1), GameControllerType(1), VendorID(0x045E), ProductID(0x028E) i> 00004480 SDL OnControllerDeviceAdded: Added at index 1.`
> 
> Magic-Ns adapter is itself visible as a controller

That's interesting both controllers are working on coop mode i just test it the problem is for Story mode only.

**Comment by theonejedi** [comment]
I am having what I suspect is the same issue. When playing gears of war, if I press y to sign in, it will instantly sign in 2 profiles, even if only only connected. I have tried changing HID to xinput but the same result. I am using an 8bitdo wireless controller.

**Comment by InterClaw** [comment]
I was having this problem and solved it by setting:
```
[HID]
hid = "sdl"
```
I'm not sure what Xenia and the game are picking up, but this is what Windows sees at least:

![image](https://github.com/xenia-canary/xenia-canary/assets/48981208/09a9a97d-e018-4b33-9a0c-4cadc4e85512)

I _am_ using [HidHide](https://github.com/nefarius/HidHide) though to get rid of some devices and make sure my controller can get Xinput slot 1. It's a DualSense emulating an Xbox 360 controller in DS4Windows. I also have some Razer peripherals that normally show up that I either hide with HidHide or disable in Device Manager.

The other Gears games don't do this though, so there `hid = "any"` works fine.

**Comment by RareMv** [comment]
In the .toml.config file you'll find:

> max_signed_profiles = 2                           	# Limits how many profiles can be assigned. Possible values: 1-4

Depending if you want to play single player or 2 player local split-screen, set this to 1 or 2 where appropriate.

**Comment by SutchAwson** [comment]
My Solution is:  Connect your controller to PC only after start the game.

Here is my finding: 
the Xenia has a default winkey setting for keyboard, like key(x) for start, key(;) for A, key(') for B
So if you unplug the controller, and start the Gears Of War 2, you can press key(x) to  go main menu, then you'll find only one User profile
<img width="129" height="64" alt="Image" src="https://github.com/user-attachments/assets/41d8c136-cd61-49a4-9842-e2e598d96f15" />
if you plug your controller now, there still be one profile.
